### PR TITLE
RCanvas - first RFrame attributes

### PR DESCRIPTION
**Pull request by linev**
Before I code too far away 

This is first showcase, how new attributes will be introduced to `RFrame`.

First important change. `RFrame` is normal drawable and will be inserted into list of primitives. 
In the future we could enforce `RFrame` have to be always on the first place (to be discuss)

`RFrame` is not required to draw histogram. All defaults can be obtained via `RStyle`, therefore object can be created on the client side and all necessary attributes  can be obtained.

As first members of `RFrame` I introduce `RAttrMargin` and `RAttrAxis` attributes.
`RAttrMargin` contains margins from pad borders and can be defined as `RPadLength` without user components. In the CSS one could use syntax like: `frame { margin_left: 0.1 + 10px; }`  
As Idea: one also can allow something like: `frame { margin: 0; }`  to set all sides at same time

`RAttrAxis` are graphical axis attributes plus min/max range, lin/log scale, invert flag, ...
For now only line attributes are really used. 

Idea for next steps with `RAttrAxis`.
When min/max values not specified, axis ranges defined by first drawable like histogram or graph.
One should be able to set only min or only max value. Also visible range should be able to specify same way. 

You can see how code will look like in new macro: `draw_frame.cxx`.  It drawing in two sub-pads. On first sub-pad RFrame configured programatically, on second sub-pad - via CSS file.

### CODE REVIEW

File: tutorials/v7/draw_frame.cxx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+   auto draw1 = subpads[0][0]->Draw(pHist);
+   draw1->AttrLine().SetColor(RColor::kRed);
+
+   // create frame before drawing histograms
+   auto frame2 = subpads[1][0]->GetOrCreateFrame();
+
+   auto draw2 = subpads[1][0]->Draw(pHist);
+   draw2->AttrLine().SetColor(RColor::kBlue).SetWidth(12);
+
+   style->ParseString("frame { margin_left: 0.3; margin_right: 0.3; x_line_color_name: blue; y_line_color_name: green; }");
```
Comment (Axel-Naumann) [inline tutorials/v7/draw_frame.cxx]: I think `_name` shouldn't be necessary - we should be able to parse colors by name or `#rrggbb` or `#rrggbbaa` or #rgb` or `#rgba`. That's independent of this PR of course.

File: tutorials/v7/draw_frame.cxx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+   frame1->Margins().SetLeft(0.2_normal);
+   frame1->Margins().SetRight(0.2_normal);
+
+   frame1->AttrX().AttrLine().SetColor(RColor::kGreen);
+   frame1->AttrY().AttrLine().SetColor(RColor::kBlue);
+
+   auto draw1 = subpads[0][0]->Draw(pHist);
+   draw1->AttrLine().SetColor(RColor::kRed);
+
+   // create frame before drawing histograms
```
Comment (Axel-Naumann) [inline tutorials/v7/draw_frame.cxx]: That's a bit surprising - I believe it's the pad's job to ensure that the frame is first.

File: tutorials/v7/draw_frame.cxx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+   auto frame1 = subpads[0][0]->GetOrCreateFrame();
+   frame1->AttrFill().SetColor(RColor::kBlue);
+   frame1->AttrBorder().SetColor(RColor::kRed);
+   frame1->AttrBorder().SetWidth(3);
+   frame1->Margins().SetTop(0.3_normal);
+   frame1->Margins().SetBottom(0.1_normal);
+   frame1->Margins().SetLeft(0.2_normal);
+   frame1->Margins().SetRight(0.2_normal);
+
+   frame1->AttrX().AttrLine().SetColor(RColor::kGreen);
```
Comment (couet) [inline tutorials/v7/draw_frame.cxx]: After an offline discussion with Sergey I understand that AttrX and AttrY are the attributes of the X and Y axis. It is not clear it is about "Axis".. I would rather call them AttrXaxis and AttrYaxis

File: tutorials/v7/draw_frame.cxx (hunk 2 above)
Comment (linev) [reply tutorials/v7/draw_frame.cxx]: In general - RFrame is just one more RDrawable in the list of primitives. And if one wants to place RText behind RFrame, such text should be before frame in list of primitives.

File: tutorials/v7/draw_frame.cxx (hunk 1 above)
Comment (linev) [reply tutorials/v7/draw_frame.cxx]: Yes, `_name` can be avoided in the future. One have to provide special text parsing for the RColor, which can distinguish if string is rgb value or color name or maybe index from palette.

File: tutorials/v7/draw_frame.cxx (hunk 2 above)
Comment (Axel-Naumann) [reply tutorials/v7/draw_frame.cxx]: The tutorial suggests that one needs to "create the frame before drawing the histogram". Why? Isn't that done automatically? What happens if the frame isn't created explicitly?

My comment is really about the lines 68, 69 in the tutorial.

File: tutorials/v7/draw_frame.cxx (hunk 3 above)
Comment (Axel-Naumann) [reply tutorials/v7/draw_frame.cxx]: What's a "xaxis"? ;-) We call this "XAxis" in ROOT7. I think `AttrX` is good enough:
- at least I guessed it right ;-)
- we'll get used to it.

File: tutorials/v7/draw_frame.cxx (hunk 2 above)
Comment (linev) [reply tutorials/v7/draw_frame.cxx]: RFrame now is only container for different attributes - all optional.
If RFrame not there - default values will be used by client.

Comment (Axel-Naumann) [reply tutorials/v7/draw_frame.cxx]: That's a very opaque interface, not obvious at all. See how I was unable to guess that. I also don't see why I cannot have e.g. `RLine` attributes without a `RFrame`.

**Pull request merged**